src/Failure.php

```diff
@@ -0,0 +1,94 @@
+<?php
+namespace Particle\Validator;
+
+class Failure
```

Comment:
`Error` implies that something is *critically* wrong. To me, `Failure` implies that something went wrong but certainly not critical, and certainly expected. After all, we're building a validation library, so having an invalid value is not exceptional: it's what the library is there for ;)